File: app/src/main/java/com/jokysss/mirror/PageFragment.java
package com.jokysss.mirror;


import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewStub;


/**
 * A simple {@link Fragment} subclass.
 * Use the {@link PageFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class PageFragment extends Fragment {
    private static final String ARG_VIEWRESID = "ViewResId";
    private int mViewResId;

    public PageFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param viewResId Parameter 1.
     * @return A new instance of fragment PageFragment.
     */
    public static PageFragment newInstance(@LayoutRes int viewResId) {
        PageFragment fragment = new PageFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_VIEWRESID, viewResId);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mViewResId = getArguments().getInt(ARG_VIEWRESID);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_page, container, false);
        ViewStub stub = view.findViewById(R.id.vs_stub);
        stub.setLayoutResource(mViewResId);
        stub.inflate();
        return view;
    }

}

File: app/src/main/java/com/jokysss/mirror/PageFragment.kt
package com.jokysss.mirror


import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.ViewStub


/**
 * A simple [Fragment] subclass.
 * Use the [PageFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class PageFragment : androidx.fragment.app.Fragment() {
    private var mViewResId: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mViewResId = arguments!!.getInt(ARG_VIEWRESID)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_page, container, false)
        val stub = view.findViewById<ViewStub>(R.id.vs_stub)
        stub.layoutResource = mViewResId
        stub.inflate()
        return view
    }

    companion object {
        private val ARG_VIEWRESID = "ViewResId"

        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param viewResId Parameter 1.
         * @return A new instance of fragment PageFragment.
         */
        fun newInstance(@LayoutRes viewResId: Int): PageFragment {
            val fragment = PageFragment()
            val args = Bundle()
            args.putInt(ARG_VIEWRESID, viewResId)
            fragment.arguments = args
            return fragment
        }
    }

}// Required empty public constructor
